packages/server-core/src/route/route/route.test.ts

```diff
@@ -0,0 +1,104 @@
+import appRootPath from 'app-root-path'
+import assert from 'assert'
+import fs from 'fs'
+import path from 'path'
+import { v4 as uuid } from 'uuid'
+
+import { destroyEngine } from '@etherealengine/engine/src/ecs/classes/Engine'
+
+import { Application } from '../../../declarations'
+import { createFeathersExpressApp } from '../../createApp'
+import { deleteFolderRecursive } from '../../util/fsHelperFunctions'
+
+const params = { isInternal: true } as any
+
+const cleanup = async (app: Application, projectName: string) => {
+  const projectDir = path.resolve(appRootPath.path, `packages/projects/projects/${projectName}/`)
+  deleteFolderRecursive(projectDir)
+  try {
+    await app.service('project').Model.destroy({ where: { name: projectName } })
+  } catch (e) {}
+}
+
+const updateXREngineConfigForTest = (newProject: string) => {
+  const testXREngineConfig = `
+  import type { ProjectConfigInterface } from '@etherealengine/projects/ProjectConfigInterface'
+
+  const config: ProjectConfigInterface = {
+    onEvent: undefined,
+    thumbnail: '/static/etherealengine.png',
+    routes: {
+      test: {
+        component: () => import('@etherealengine/client/src/pages/index'),
+      },
+      "${newProject}": {
+        component: () => import('@etherealengine/client/src/pages/index'),
+      }
+    },
+    services: undefined,
+    databaseSeed: undefined
+  }
+  
+  export default config
+  `
+
+  const projectsRootFolder = path.join(appRootPath.path, 'packages/projects/projects/')
+  const projectLocalDirectory = path.resolve(projectsRootFolder, newProject)
+  const xrEngineConfigFilePath = path.resolve(projectLocalDirectory, 'xrengine.config.ts')
+
+  fs.rmSync(xrEngineConfigFilePath)
+  fs.writeFileSync(xrEngineConfigFilePath, testXREngineConfig)
+}
+
+describe('route.test', () => {
+  let app: Application
+  let newProject: string
+
+  before(async () => {
+    app = createFeathersExpressApp()
+    await app.setup()
+  })
+
+  after(async () => {
+    await destroyEngine()
+    await cleanup(app, newProject)
+  })
+
+  it('should find the installed project routes', async () => {
+    newProject = `test-project-${uuid()}`
+    await app.service('project').create({ name: newProject }, params)
+    updateXREngineConfigForTest(newProject)
```

Comment:
let's define a different value rather than reusing newProject here

Reply:
I have used `testRoute` now.

Reply:
We want two different variables, one for the project name and one for the route

Reply:
Understood. I have used 2 variables - `testProject` and `testRoute` having different values and testing them.